engine/src/it/java/org/pentaho/di/job/JobExportIT.java

```diff
@@ -0,0 +1,180 @@
+    deleteFolder( new File( EXPORT_DIR ) );
+    new File( EXPORT_DIR ).mkdirs();
+
+    KettleEnvironment.init();
+  }
+
+  @AfterClass
+  public static void tearDownOnce() {
+    deleteFolder( new File( EXPORT_DIR ) );
+  }
+
+  @Before
+  public void setUp() throws KettleException {
+    KettleFileRepositoryMeta repositoryMeta = new KettleFileRepositoryMeta( FILE_REPOSITORY_ID, FILE_REPOSITORY_NAME,
+      FILE_REPOSITORY_DESC, REPOSITORY_ROOT_DIR );
+    repository = new KettleFileRepository();
+    repository.init( repositoryMeta );
+    repository.connect( null, null );
+  }
+
+  @After
+  public void tearDown() {
+    repository.disconnect();
+  }
+
+  /**
+   * Given a Job located in non-root directory of a repository,
+   * and referencing to a Transformation using 'Internal.Entry.Current.Directory' variable.
+   * <br/>
+   * When this Job is exported into a zip file,
+   * then the referenced Transformation should be exported as well.
+   */
+  @Test
+  public void shouldExportJobAndRelatedTransformationFile() throws IOException, MetaStoreException, KettleException {
+    RepositoryDirectoryInterface repositoryDir = repository.loadRepositoryDirectoryTree();
+    repositoryDir = repositoryDir.findDirectory( REPOSITORY_DIR );
+
+    JobMeta jobMeta = repository.loadJob( JOB_NAME, repositoryDir, null, null );
+    Job job = new Job( repository, jobMeta );
+
+    ResourceUtil.serializeResourceExportInterface( EXPORT_FILE, job.getJobMeta(), job, repository, null );
+
+    File zipFile = new File( EXPORT_FILE );
+    assertTrue( zipFile.exists() );
+
+    extractZip( zipFile, EXTRACT_DIR );
+
+    // assert that either of files, job and transformation, have been exported.
+    assertTrue( new File( EXTRACTED_JOB_FILE ).exists() );
+    assertTrue( new File( EXTRACTED_TRANS_FILE ).exists() );
+  }
+
+  private static void deleteFolder( File folder ) {
+    File[] files = folder.listFiles();
+    if ( files != null ) { //some JVMs return null for empty dirs
+      for ( File f : files ) {
+        if ( f.isDirectory() ) {
+          deleteFolder( f );
+        } else {
+          f.delete();
+        }
+      }
+    }
+    folder.delete();
+  }
+
+  private static void extractZip( File zipFile, String extractDir ) throws IOException {
+    int BUFFER = 2048;
+    ZipFile zip = new ZipFile( zipFile );
+
+    new File( extractDir ).mkdir();
+    Enumeration zipFileEntries = zip.entries();
+
+    // Process each entry
+    while ( zipFileEntries.hasMoreElements() ) {
+      // grab a zip file entry
+      ZipEntry entry = (ZipEntry) zipFileEntries.nextElement();
+      String currentEntry = entry.getName();
+
+      File destFile = new File( extractDir, currentEntry );
+      File destinationParent = destFile.getParentFile();
+
+      // create the parent directory structure if needed
+      destinationParent.mkdirs();
+
+      if ( !entry.isDirectory() ) {
+        BufferedInputStream is = new BufferedInputStream( zip
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *is*. The resource is referenced at lines 144, 159, 171. The resource is closed at line 176. There are other execution paths that do not contain closure statements, for example, when BufferedInputStream.read throws an exception. Ensure *is* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).